lib/api/controllers/funnelController.js

```diff
@@ -200,16 +201,19 @@ class FunnelController {
       return -1;
     }
 
-    let result = 0;
+    // if the connection is closed there is no need to execute the request
+    // => discarding it
+    if (!this.kuzzle.router.isConnectionAlive(request.context)) {
+      delete this.pendingRequests[request.id];
+      return 0;
+    }
+
     this.checkRights(request)
       .then(modifiedRequest => this.processRequest(modifiedRequest))
       .then(processResult => callback(null, processResult))
-      .catch(err => {
-        this._executeError(err, request, true, callback);
-        result = 1;
-      });
+      .catch(err => this._executeError(err, request, true, callback));
```

Comment:
I agree that for now, the implemented use cases only need to distinguish `-1` from other result values...
... but the actual behaviour of this method just is not consistent with the jsdoc.

IMHO, we need
* either to update the jsdoc (and return only `0` or `-1` - that means also return `0` in line 197 above)
* or to return `1` each time we have an error, that means, for example, replace the current block code by somthing like:
```javascript
return this.checkRights(request)
  .then(modifiedRequest => this.processRequest(modifiedRequest))
  .then(processResult => {
    callback(null, processResult);
    return 0;
  })
  .catch(err => {
    this._executeError(err, request, true, callback);
    return 1;
  });
```